### ISSUE TITLE
JOSS review - Functionality documentation

### ISSUE THREAD

**Issue by craig-willis** [terminological/dtrackr#14]
Review issue: https://github.com/openjournals/joss-reviews/issues/4707
Branch reviewed: release-0.2.4

**Problem**
It's not immediately clear to me from the documentation which `dplyr` and `tidyr` operations are supported by `dtrackr` or why some are missing. This can be determined only by comparing the list of `p_*` functions and inspecting the source code comments.

I'm not a regular user of `dplyr`, but based on comparison to https://dplyr.tidyverse.org/reference/ see the following operations are not supported by `dtrackr`: `count`, `tally`, `glimpse`, `group*`, `pull`, `summarize`, `slice*`, `bind_cols`, `union_all`, `nest_join`, `rowwise`.

Code comments indicate `count / tally - not included as summarisation step, not generally part of a dplyr pipeline` and `group_map/group_walk -- not included as does not return DF`.

The `pivot_wider` and `pivot_longer` appear to be the only operations supported from `tidyr`.

As a user, it would be helpful to have a quick reference of supported operations and explanations of why certain operations are/are not included. This might also be an opportunity to highlight the relationship between `dplyr::filter` and `exclude_all`, since it's not strictly wrapping the `dplyr` operation.

**Comment by robchallen** [comment]
So I should have addressed this issue first.

There were some gaps in functionality support. These were partly pragmatic and partly due to me not using those functions very much. Part of the implementation though is achieved by sub-classing `data.frame` (as `trackr_df`) and implementing the wrapper functions as S3 methods for the subclass. This means that mixing the native `dplyr` and `dtrackr` functions should just work and can be switched back and forth, and in theory implementing all the `dplyr` methods is not strictly required.

Anyway in the light of this comment I implemented the missing functions in `dplyr`. This triggered me to also address some of the testing issues and improve test coverage (see #13). I have now included `dtrackr` functions for `slice*`, `bind_cols`, `union_all`, `nest_join`.

For documentation of the supported reference I have updated the [function reference page](https://terminological.github.io/dtrackr/reference/index.html) to make it more logically organised and providing some detail about the rationale for the scope. This includes an explanation about what other functions are out of scope, and not yet in scope, compared to `dplyr`.
